crates/sncast/src/helpers/fee.rs

```diff
@@ -92,44 +95,44 @@ impl FeeArgs {
                         bail!("--max-fee should be greater than or equal to --max-gas-unit-price")
                     }
                     (None, _, _) => FeeSettings::Strk {
-                        max_gas: self.max_gas.map(TryIntoConv::try_into_).transpose()?,
+                        max_gas: self
+                            .max_gas
+                            .and_then(|val| NonZeroU64::try_from_(Felt::from(val)).ok()),
                         max_gas_unit_price: self
                             .max_gas_unit_price
-                            .map(TryIntoConv::try_into_)
-                            .transpose()?,
-                    },
-                    (Some(max_fee), None, Some(max_gas_unit_price)) => FeeSettings::Strk {
-                        max_gas: Some(
-                            max_fee
-                                .floor_div(&NonZeroFelt::from_felt_unchecked(max_gas_unit_price))
-                                .try_into_()?,
-                        ),
-                        max_gas_unit_price: Some(max_gas_unit_price.try_into_()?),
-                    },
-                    (Some(max_fee), Some(max_gas), None) => FeeSettings::Strk {
-                        max_gas: Some(max_gas.try_into_()?),
-                        max_gas_unit_price: Some(
-                            max_fee
-                                .floor_div(&NonZeroFelt::from_felt_unchecked(max_gas))
-                                .try_into_()?,
-                        ),
+                            .and_then(|val| NonZeroU128::try_from_(Felt::from(val)).ok()),
                     },
+                    (Some(max_fee), None, Some(max_gas_unit_price)) => {
+                        let max_gas = NonZeroFelt::try_from(Felt::from(max_fee).floor_div(&max_gas_unit_price)).context("Calculated max gas from provided --max-fee and --max-gas-unit-price is zero. Please increase --max-fee to obtain a positive gas amount")?;
+                        FeeSettings::Strk {
+                            max_gas: NonZeroU64::try_from_(Felt::from(max_gas)).ok(),
+                            max_gas_unit_price: NonZeroU128::try_from_(Felt::from(
+                                max_gas_unit_price,
+                            ))
+                            .ok(),
+                        }
+                    }
+                    (Some(max_fee), Some(max_gas), None) => {
+                        let max_gas_unit_price = NonZeroFelt::try_from(Felt::from(max_fee).floor_div(&max_gas)).context("Calculated max gas unit price from provided --max-fee and --max-gas is zero. Please increase --max-fee or decrease --max-gas to ensure a positive gas unit price")?;
```

Comment:
```suggestion
                        let max_gas_unit_price = NonZeroFelt::try_from(Felt::from(max_fee).floor_div(&max_gas)).context("Calculated max gas unit price from provided --max-fee and --max-gas is 0. Please increase --max-fee or decrease --max-gas to ensure a positive gas unit price")?;
```

I think it makes more sense to use decimal value here and in other places.

Reply:
Or even `= 0` but not sure about that